src/core/Makefile.am

```diff
@@ -1,11 +1,29 @@
 lib_LTLIBRARIES = libdyad_core.la
-libdyad_core_la_SOURCES = dyad_core.c
-libdyad_core_la_LIBADD = $(top_builddir)/src/utils/libutils.la $(top_builddir)/src/utils/libmurmur3.la $(FLUX_CORE_LIBS)
-libdyad_core_la_CPPFLAGS = $(AM_CPPFLAGS) -I$(top_builddir)/src/utils $(FLUX_CORE_CFLAGS)
-libdyad_core_la_LDFLAGS = -export-symbols dyad_core.sym $(AM_LDFLAGS)
+libdyad_core_la_SOURCES = \
+	dyad_core.c
+libdyad_core_la_LIBADD = \
+	$(top_builddir)/src/dtl/libdyad_dtl.la \
+	$(UCX_LIBS) \
+	$(JANSSON_LIBS) \
+	$(FLUX_CORE_LIBS)
+libdyad_core_la_CFLAGS = \
+	$(AM_CFLAGS) \
+	-I$(top_srcdir)/src/utils \
```

Comment:
I absolutely agree with you @hariharan-devarajan, but I did this because we wanted to stay (relatively) consistent with Flux. This `flux-core` repo does something similar to this in its `Makefile.am`s.

However, regardless of what Flux does and how much we want to follow that, I also believe that any changes we might want to make regarding organization of files should belong in a separate PR.

Reply:
I agree that we should put more effort to clean up installation and make distinction between public headers and private headers. This can be worked on in a followup PR.